synapse/handlers/room.py

```diff
@@ -1038,12 +1054,18 @@ def filter_evts(events):
 
 
 class RoomEventSource(object):
-    def __init__(self, hs):
+    def __init__(self, hs: "HomeServer"):
         self.store = hs.get_datastore()
 
     async def get_new_events(
-        self, user, from_key, limit, room_ids, is_guest, explicit_room_id=None
-    ):
+        self,
+        user: UserID,
+        from_key: str,
+        limit: int,
+        room_ids: List[str],
+        is_guest: bool,
+        explicit_room_id: Optional[str] = None,
```

Comment:
Looks like these last two arguments aren't even used anymore.

Reply:
It's part of an interface for the notifier

Reply:
Ah, fair